src/components/SelectField/SelectField.js

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable max-len */
+/* eslint-env browser */
+import React, { Component } from 'react';
+import propTypes from 'prop-types';
+
+
+class SelectField extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      valid: null,
+      message: '',
+      value: this.getSelectedOption(),
+      showErrorMessage: this.props.showErrorMessage,
+    };
+    this.setRef = (element) => {
+      this.inputRef = element;
+    };
+    this.validateField = this.validateField.bind(this);
+    this.onChangeHandler = this.onChangeHandler.bind(this);
+  }
+
+  /**
+   * Validate initial state
+   * (will trigger an update through the validateField function)
+   */
+  componentDidMount() {
+    this.validateField();
+  }
+
+  /**
+   * When component has updated send state to parent
+   */
+  componentDidUpdate() {
+    this.sendStateToParent();
+  }
+
+  /**
+   * Handle the onChange and onBlur events
+   * @param e
+   */
+  onChangeHandler(e) {
+    const value = e.target.value;
+    this.setState({
+      value,
+      showErrorMessage: true,
+    });
+    this.validateField(e);
+  }
+
+  /**
+   * Get the default option, marked as selected from the props.options array
+   * Returns the value or undefined
+   */
+  getSelectedOption() {
+    let selected = this.props.options.find(item => item.selected === true);
```

Comment:
check if find() is support by IE10/IE9 sure there was some issue with certain version using find()

Reply:
good point 👍